resources/run_sampling.rb

```diff
@@ -306,38 +306,36 @@ def write_csv(sample_results, outfile)
   end
 end
 
-if __FILE__ == $PROGRAM_NAME
-  options = {}
-  OptionParser.new do |opts|
-    opts.banner = "Usage: #{File.basename(__FILE__)} -p project_name -n num_datapoints -o outfile\n e.g., #{File.basename(__FILE__)} -p project_national -n 10000 -o buildstock.csv"
+options = {}
+OptionParser.new do |opts|
+  opts.banner = "Usage: #{File.basename(__FILE__)} -p project_name -n num_datapoints -o outfile\n e.g., #{File.basename(__FILE__)} -p project_national -n 10000 -o buildstock.csv"
 
-    opts.on('-p', '--project <STRING>', 'Project Name') do |t|
-      options[:project] = t
-    end
-
-    opts.on('-n', '--num-datapoints <INTEGER>', 'Number of datapoints') do |t|
-      options[:numdps] = t.to_i
-    end
+  opts.on('-p', '--project <STRING>', 'Project Name') do |t|
+    options[:project] = t
+  end
 
-    opts.on('-o', '--output <STRING>', 'Output file name') do |t|
-      options[:outfile] = t
-    end
+  opts.on('-n', '--num-datapoints <INTEGER>', 'Number of datapoints') do |t|
+    options[:numdps] = t.to_i
+  end
 
-    opts.on_tail('-h', '--help', 'Display help') do
-      puts opts
-      exit
-    end
-  end.parse!
+  opts.on('-o', '--output <STRING>', 'Output file name') do |t|
+    options[:outfile] = t
+  end
 
-  if (not options[:project]) || (not options[:numdps]) || (not options[:outfile])
-    fail "ERROR: All 3 arguments are required. Call #{File.basename(__FILE__)} -h for usage."
+  opts.on_tail('-h', '--help', 'Display help') do
+    puts opts
+    exit!
```

Comment:
I also changed this from `exit` to `exit!` so that it doesn't display a stacktrace.

Before:
```
shorowit@SHOROWIT-36132S:/mnt/c/git/resstock/resources$ openstudio run_sampling.rb --help
Usage: run_sampling.rb -p project_name -n num_datapoints -o outfile
 e.g., run_sampling.rb -p project_national -n 10000 -o buildstock.csv
    -p, --project <STRING>           Project Name
    -n, --num-datapoints <INTEGER>   Number of datapoints
    -o, --output <STRING>            Output file name
    -h, --help                       Display help
Error executing argv: []
Error: exit in /mnt/c/git/resstock/resources/run_sampling.rb:327:in `exit'
/mnt/c/git/resstock/resources/run_sampling.rb:327:in `block (2 levels) in <top (required)>'
:/ruby/2.7.0/optparse.rb:1589:in `block in parse_in_order'
:/ruby/2.7.0/optparse.rb:1575:in `catch'
:/ruby/2.7.0/optparse.rb:1575:in `parse_in_order'
:/ruby/2.7.0/optparse.rb:1569:in `order!'
:/ruby/2.7.0/optparse.rb:1666:in `permute!'
:/ruby/2.7.0/optparse.rb:1691:in `parse!'
/mnt/c/git/resstock/resources/run_sampling.rb:329:in `<top (required)>'
eval:134:in `require'
eval:134:in `require'
:/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
:/openstudio_cli.rb:1736:in `execute'
:/openstudio_cli.rb:804:in `execute'
:/openstudio_cli.rb:1973:in `<main>'
eval:188:in `eval'
eval:188:in `require_embedded_absolute'
eval:173:in `block in require_embedded'
eval:167:in `each'
eval:167:in `require_embedded'
eval:126:in `require'
eval:3:in `<main>'
```

After:
```
shorowit@SHOROWIT-36132S:/mnt/c/git/resstock/resources$ openstudio run_sampling.rb --help
Usage: run_sampling.rb -p project_name -n num_datapoints -o outfile
 e.g., run_sampling.rb -p project_national -n 10000 -o buildstock.csv
    -p, --project <STRING>           Project Name
    -n, --num-datapoints <INTEGER>   Number of datapoints
    -o, --output <STRING>            Output file name
    -h, --help                       Display help
```